holoviews/core/util.py

```diff
@@ -608,6 +608,23 @@ def python2sort(x,key=None):
     return itertools.chain.from_iterable(sorted(group, key=key) for group in groups)
 
 
+def merge_dimensions(dimensions_list):
+    """
+    Merges lists of fully or partially overlapping dimensions by
+    combining their values.
+    """
```

Comment:
The code is clear enough, but it might be nice to add a doctest example here (if we are using doctests).